src/datumaro/plugins/transforms.py

```diff
@@ -1447,3 +1448,90 @@ def __iter__(self):
                         )
                     updated_anns.append(new_ann)
                 yield item.wrap(annotations=updated_anns)
+
+
+class AstypeAnnotations(ItemTransform):
+    """
+    Enables the conversion of annotation types for the categories and individual items within a dataset.|n
+    |n
+    Based on a specified mapping, it transforms the annotation types,|m
+    changing them to 'Label' if they are categorical, and to 'Caption' if they are of type string, float, or integer.|n
+    |n
+    Examples:|n
+        - Convert type of `title` annotation|n
+
+        .. code-block::
+
+        |s|s%(prog)s --mapping 'title:Caption'
+    """
+
+    @staticmethod
+    def _split_arg(s):
+        columns = s.split(",")
+        results = []
+        for column in columns:
+            parts = column.split(":")
+            if len(parts) != 2:
+                raise argparse.ArgumentTypeError()
+            results.append((parts[0], parts[1]))
+        return results
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument(
+            "--mapping",
+            action="append",
+            type=cls._split_arg,
+            dest="mapping",
+            help="Annotations type in the form of: '<src>:<dst>' (repeatable)",
+        )
+        return parser
+
+    def __init__(
+        self,
+        extractor: IDataset,
+        mapping: Optional[Union[Dict[str, str], List[Tuple[str, str]]]] = None,
+    ):
+        super().__init__(extractor)
+
+        # Turn off for default setting
+        # assert isinstance(mapping, (dict, list))
+
+        if isinstance(mapping, list):
+            mapping = dict(mapping)
+
+        self._categories = {}
+
+        src_categories = self._extractor.categories()
+        src_tabular_cat = src_categories.get(AnnotationType.tabular)
+        self._tabular_cat_types = {}
+
+        # Make LabelCategories
+        self._id_mapping = {}
+        dst_label_cat = LabelCategories()
+        if src_tabular_cat is not None:
+            for src_cat in src_tabular_cat:
+                if src_cat.dtype == CategoricalDtype():
+                    dst_parent = src_cat.name
+                    dst_labels = sorted(src_cat.labels)
+                    for dst_label in dst_labels:
+                        dst_index = dst_label_cat.add(dst_label, parent=dst_parent, attributes={})
+                        self._id_mapping[dst_label] = dst_index
+                    dst_label_cat.add_label_group(src_cat.name, src_cat.labels, group_type=0)
+                self._tabular_cat_types[src_cat.name] = src_cat.dtype
+            self._categories[AnnotationType.label] = dst_label_cat
```

Comment:
```suggestion
        if src_tabular_cat is None:
            self._categories[AnnotationType.label] = dst_label_cat
            return
        for src_cat in src_tabular_cat:
            if src_cat.dtype != CategoricalDtype(): continue
            dst_parent = src_cat.name
            dst_labels = sorted(src_cat.labels)
            for dst_label in dst_labels:
                dst_index = dst_label_cat.add(dst_label, parent=dst_parent, attributes={})
                self._id_mapping[dst_label] = dst_index
            dst_label_cat.add_label_group(src_cat.name, src_cat.labels, group_type=0)
            self._tabular_cat_types[src_cat.name] = src_cat.dtype
        self._categories[AnnotationType.label] = dst_label_cat
```
to avoid nested if-else statement.

Reply:
Fix this in [ba98d27](https://github.com/openvinotoolkit/datumaro/pull/1484/commits/ba98d2716972353e95a34293bc012c517edbaf3d)